internal/controller/provisioner/objects.go

```diff
@@ -164,7 +179,11 @@ func (p *NginxProvisioner) buildNginxResourceObjects(
 	}
 	objects = append(objects, service, deployment)
 
-	return objects, err
+	if len(errs) > 0 {
+		return objects, errors.Join(errs...)
+	}
+
+	return objects, nil
```

Comment:
I think you could just do `return objects, errors.Join(errs...)`. If errs is empty, it should return nil.